content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,794 @@
++++
+title = "0.8 to 0.9"
+weight = 5
+sort_by = "weight"
+template = "book-section.html"
+page_template = "book-section.html"
+insert_anchor_links = "right"
+[extra]
+long_title = "Migration Guide: 0.8 to 0.9"
++++
+
+Before migrating make sure to run rustup update
+
+Bevy relies heavily on improvements in the Rust language and compiler. As a result, the Minimum Supported Rust Version (MSRV) is "the latest stable release" of Rust.
+
+### [Make `Resource` trait opt-in, requiring `#[derive(Resource)]` V2](https://github.com/bevyengine/bevy/pull/5577)
+
+Add `#[derive(Resource)]` to all types you are using as a resource.
+
+If you are using a third party type as a resource, wrap it in a tuple struct to bypass orphan rules. Consider deriving `Deref` and `DerefMut` to improve ergonomics.
+
+`ClearColor` no longer implements `Component`. Using `ClearColor` as a component in 0.8 did nothing.
+Use the `ClearColorConfig` in the `Camera3d` and `Camera2d` components instead.
+
+### [Plugins own their settings. Rework PluginGroup trait.](https://github.com/bevyengine/bevy/pull/6336)
+
+The `WindowDescriptor` settings have been moved from a resource to `WindowPlugin::window`:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(WindowDescriptor {
+    width: 400.0,
+    ..default()
+  })
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(WindowPlugin {
+  window: WindowDescriptor {
+    width: 400.0,
+    ..default()
+  },
+  ..default()
+}))
+```
+
+The `AssetServerSettings` resource has been removed in favor of direct `AssetPlugin` configuration:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(AssetServerSettings {
+    watch_for_changes: true,
+    ..default()
+  })
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(AssetPlugin {
+  watch_for_changes: true,
+  ..default()
+}))
+```
+
+`add_plugins_with` has been replaced by `add_plugins` in combination with the builder pattern:
+
+```rust
+// Old (Bevy 0.8)
+app.add_plugins_with(DefaultPlugins, |group| group.disable::<AssetPlugin>());
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.build().disable::<AssetPlugin>());
+```
+

```

Comment:
```suggestion

`PluginGroupBoulder` and the `PluginGroup` trait have also been reworked.

```rust
// Old (Bevy 0.8)
impl PluginGroup for HelloWorldPlugins {
    fn build(&mut self, group: &mut PluginGroupBuilder) {
        group.add(PrintHelloPlugin).add(PrintWorldPlugin);
    }
}

// New (Bevy 0.9)
impl PluginGroup for HelloWorldPlugins {
    fn build(self) -> PluginGroupBuilder {
        PluginGroupBuilder::start::<Self>()
            .add(PrintHelloPlugin)
            .add(PrintWorldPlugin)
    }
}

```

This also includes changes from 6411 (the `::start::<Self>()` bit) which is not otherwise mentioned in the migration guide. But the two are sort of intermingled and the title of this section mentions PluginGroup, so I thought I'd just cram this in here.